src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/DraftApprenticeshipController.cs

```diff
@@ -428,12 +436,44 @@ public async Task<IActionResult> AddDraftApprenticeship(string changeCourse, str
             });
         }
 
+        [HttpGet]
+        [Route("{DraftApprenticeshipHashedId}/edit", Name = RouteNames.DraftApprenticeshipEdit)]
+        [Authorize(Policy = nameof(PolicyNames.HasContributorOrAbovePermission))]
+        [ServiceFilter(typeof(UseCacheForValidationAttribute))]
+        public async Task<IActionResult> EditDraftApprenticeship(DraftApprenticeshipRequest request)
+        {
+            try
+            {
+                var model = await _modelMapper.Map<IDraftApprenticeshipViewModel>(request);
+
+                if (model is EditDraftApprenticeshipViewModel editModel)
+                {
+                    ApplyLearnerDataSyncUpdates(editModel);
+
+                    await AddLegalEntityAndCoursesToModel(editModel);
+                    PrePopulateDates(editModel);
```

Comment:
Should this functionality ApplyLearnerDataSyncUpdates, AddLegalEntityAndCoursesToModel, PrePopulateDates
be in the Mapper?